include/ck/tensor_operation/gpu/device/device_convnd_fwd_xdl_nhwc_kyxc_nhwk.hpp

```diff
@@ -862,17 +862,11 @@ struct DeviceConvNDFwdXdl_Input_N_Hi_Wi_C_Weight_K_Y_X_C_Output_N_Ho_Wo_K
     static bool IsSupportedArgument(const Argument& arg)
     {
         // Input tensors can't be bigger than 2GB each.
-        constexpr std::size_t GB2 = 2 * 1e9;
+        constexpr ck::long_index_t GB2 = (ck::long_index_t{1} << 31);
 
-        if(arg.a_grid_desc_k0_m_k1_.GetElementSpaceSize() > GB2)
-        {
-            return false;
-        }
-        if(arg.b_grid_desc_k0_n_k1_.GetElementSpaceSize() > GB2)
-        {
-            return false;
-        }
-        if(arg.c_grid_desc_m_n_.GetElementSpaceSize() > GB2)
+        if(arg.a_grid_desc_k0_m_k1_.GetElementSpaceSize() * sizeof(ADataType) > GB2 ||
+           arg.b_grid_desc_k0_n_k1_.GetElementSpaceSize() * sizeof(BDataType) > GB2 ||
+           arg.c_grid_desc_m_n_.GetElementSpaceSize() * sizeof(CDataType) > GB2)
```

Comment:
why the factor sizeof(...)?

Reply:
Me thinks the safety check should be on the memory size not number of elements - that was the intention of Chao's fix. Could you confirm? @asroy

Reply:
GetElementSpaceSize returns the 1d space the tensor occupies (in unit of element, not byte), so we need to add sizeof() here

Reply:
The 2GB limit is because of the range of int32_t/index_t, and has nothing to day with the occupied memory space. Also GetElementSpaceSize is a property of tensor descriptor, which can be used to any data types. so I think we don't need add sizeof() here.

Reply:
[594a4c2](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/212/commits/594a4c22d129e4eac6930355ea180a0e156735e5)

Reply:
Unfortunately the 2GB limit is due to "offset" in instruction is uint32_t, and since we use int32_t for offset in C++, only 31bit is usable, thus the 2GB limit.

Also GetElementSpaceSize is not taking into consideration of datatype of tensor
@j4yan